### PR TITLE
badgeV1

### PR DESCRIPTION
**Pull request by lukeac123**
- rename "BadgeContent" to "value"
- inline styling when no child is passed to the component
- truncation on numbers > 4 characters or when max value is exceeded
- truncation on strings > 4 characters

### CODE REVIEW

**Comment by lukeac123** [comment]
- Waiting for final sign off from design
- once approved will add a changeset and merge to core

File: packages/lab/src/badge/Badge.css [hunk 1]
```diff
@@ -1,67 +1,34 @@
...
-  margin-right: var(--badge-offset-inline);
+  align-items: center;
+  justify-content: center;
+  display: inline-flex;
+  margin: auto;
 
-  padding: var(--badge-content-padding);
-  background: var(--badge-content-icon-color);
+  font-size: var(--salt-accent-fontSize);
+  font-family: var(--salt-text-fontFamily);
```
Comment (libertymayc) [inline packages/lab/src/badge/Badge.css:24]:
```suggestion
```
defined globally so we don't need it here

File: packages/lab/src/badge/Badge.css [hunk 2]
```diff
@@ -1,67 +1,34 @@
...
 
+.saltBadge-topRight {
   position: relative;
-
-  flex-wrap: wrap;
-  align-items: center;
-  align-content: center;
-  flex-direction: row;
-  justify-content: center;
+  top: -7px;
```
Comment (libertymayc) [inline packages/lab/src/badge/Badge.css:32]: these are quite specific, are they prone to breaking?

File: packages/lab/src/badge/Badge.css (hunk 1 above)
Comment (ZarkoPernar) [reply packages/lab/src/badge/Badge.css:24]: Didn't we have an issue with a project overwriting font styles with an html selector which trumps global @libertymayc ?

File: packages/lab/stories/badge/badge.stories.tsx [hunk 3]
```diff
@@ -41,29 +46,41 @@ const Template: ComponentStory<typeof Badge> = (args) => {
...
+  value: "lots",
   children: "Lorem Ipsum",
 };
 
-export const MaxNumber = Template.bind({});
-MaxNumber.args = {
-  max: 99,
-  badgeContent: 150,
+export const TruncatedString = Template.bind({});
+String.args = {
```
Comment (ZarkoPernar) [inline packages/lab/stories/badge/badge.stories.tsx:76]:
```suggestion
TruncatedString.args = {
```

File: packages/lab/stories/badge/badge.stories.tsx [hunk 4]
```diff
@@ -13,25 +13,30 @@ export const All: ComponentStory<typeof Badge> = () => {
...
+          Lorem Ipsum{" "}
         </Badge>
+        <Badge value={"hi"}> Lorem Ipsum </Badge>
+        <Badge value={"lots and lots"}> Lorem Ipsum </Badge>
+      </div>
+      <div
+        style={{
+          display: "flex",
+          flexDirection: "row",
+        }}
```
Comment (ZarkoPernar) [inline packages/lab/stories/badge/badge.stories.tsx:36]: Does it look the same without this?

File: packages/lab/src/badge/Badge.css [hunk 5]
```diff
@@ -1,67 +1,34 @@
...
-  /* default small type */
-  width: auto;
-  min-width: 15px;
-  line-height: var(--badge-content-lineHeight);
-  border-radius: var(--badge-content-borderRadius);
-  text-align: center;
+  /* Should this vary according to touch size */
+  padding-inline: var(--salt-spacing-50);
+  line-height: var(--salt-accent-lineHeight);
+  min-width: 10px;
```
Comment (ZarkoPernar) [inline packages/lab/src/badge/Badge.css:14]: As per Navs comment
```suggestion
  min-width: var(--salt-accent-lineHeight);
```

Comment (libertymayc) [reply packages/lab/src/badge/Badge.css:14]: line height on minwidth? 😛

Comment (joshwooding) [reply packages/lab/src/badge/Badge.css:14]: This is fine

File: packages/lab/stories/badge/badge.stories.tsx (hunk 4 above)
Comment (ZarkoPernar) [reply packages/lab/stories/badge/badge.stories.tsx:36]: Nope, you are right. All good 👍

File: packages/lab/src/badge/Badge.css (hunk 1 above)
Comment (libertymayc) [reply packages/lab/src/badge/Badge.css:24]: did we?

**Review by lilyvc — APPROVED** [review]

File: packages/lab/src/badge/Badge.css (hunk 2 above)
Comment (lukeac123) [reply packages/lab/src/badge/Badge.css:32]: seems to be fine on all the stories but how else would you suggest doing the positioning ?

**Pull request closed without merging**